File: app/src/main/java/email/schaal/ocreader/api/json/v12/ItemIds.java
package email.schaal.ocreader.api.json.v12;

import java.util.HashSet;
import java.util.Set;

import email.schaal.ocreader.model.Item;

/**
 * Aggregates item ids, used to mark multiple items as read
 */
public class ItemIds {
    private final Set<Long> items = new HashSet<>();

    public ItemIds(Iterable<Item> items) {
        for (Item item : items) {
            this.items.add(item.getId());
        }
    }

    public Set<Long> getItems() {
        return items;
    }
}

File: app/src/main/java/email/schaal/ocreader/api/json/v12/ItemIds.kt
package email.schaal.ocreader.api.json.v12

import com.squareup.moshi.JsonClass
import email.schaal.ocreader.api.json.Items
import email.schaal.ocreader.database.model.Item

/**
 * Aggregates item ids, used to mark multiple items as read
 */
@JsonClass(generateAdapter = true)
class ItemIds(val items: List<Long>) {
    constructor(sourceItems: Iterable<Item>) : this(sourceItems.map { it.id })
}